examples/accounts/accounts.pact

```diff
@@ -18,129 +18,129 @@
 
   (defun fund-account (address amount date)
-    (enforce-keyset 'accounts-admin-keyset)
-    (update accounts address
-            { "balance": amount
-            , "amount": amount
-            , "date": date
-            , "data": "Admin account funding" }
-      ))
+    (with-capability (ADMIN)
+      (update accounts address
+              { "balance": amount
+              , "amount": amount
+              , "date": date
+              , "data": "Admin account funding" }
+      )))
 
   (defun read-all ()
-    (map (read-account-admin) (keys accounts)))
+    (with-capability (ADMIN)
+      (map (read-account-admin) (keys accounts))))
 
   (defpact payment (payer payer-entity payee payee-entity amount date)
     "Debit PAYER at PAYER-ENTITY then credit PAYEE at PAYEE-ENTITY for AMOUNT on DATE"
     (step-with-rollback payer-entity
-      (debit payer amount date
+      (priv-debit payer amount date
             { "payee": payee
             , "payee-entity": payee-entity
             , PACT_REF: (pact-id)
             })
-      (credit payer amount date
+      (priv-credit payer amount date
            { PACT_REF: (pact-id), "note": "rollback" }))
 
     (step payee-entity
-      (credit payee amount date
+      (priv-credit payee amount date
             { "payer": payer
             , "payer-entity": payer-entity
             , PACT_REF: (pact-id)
             }
       )))
 
+  (defun priv-debit (id amt date note)
+    (with-capability (TRANSFER) (debit id amt date note)))
 
-  (defun enforce-auth (keyset:keyset auth)
-    (if (= auth AUTH_KEYSET)
-      (enforce-keyset keyset)
-      (enforce (= auth (format "%s" [(pact-id)]))
-        "Invalid access of pact account")))
-
+  (defun priv-credit (id amt date note)
+    (with-capability (TRANSFER) (credit id amt date note)))
 
   (defun debit (acct amount date data)
     "Debit AMOUNT from ACCT balance recording DATE and DATA"
-    (with-read accounts acct
-              { "balance":= balance
-              , "keyset" := ks
-              , "auth" := auth
-              }
-      (check-balance balance amount)
-      (enforce-auth ks auth)
-      (update accounts acct
+    (require-capability (TRANSFER))
+    (with-capability (USER_GUARD acct)
+      (with-read accounts acct
+                { "balance":= balance
+                }
+        (check-balance balance amount)
+        (update accounts acct
                 { "balance": (- balance amount)
                 , "amount": (- amount)
                 , "date": date
                 , "data": data
                 }
-          )))
+          ))))
 
  (defun credit (acct amount date data)
    "Credit AMOUNT to ACCT balance recording DATE and DATA"
-   (with-read accounts acct
-              { "balance":= balance }
-     (update accounts acct
-            { "balance": (+ balance amount)
-            , "amount": amount
-            , "date": date
-            , "data": data
-            }
+   (require-capability (TRANSFER))
```

Comment:
missing `with-capability`?

```suggestion
   (require-capability (TRANSFER))
   (with-capability (USER_GUARD acct)
```

Reply:
No, credits do not require signatures, only debits.